File: lotto-sim.py
#- KIRJOITTANUT HERRA KADDI -#
import random
import time
import os

maara = 43
arvattavamaara = 7
arvatutnumerot = []
oikeatnumerot = []

laskin = 1
oikeinmenneet = 0

# RETURNAA FALSE JOS ON SAMA NUMERO KUN MIKÄ ON LISTASSA JO JA KATSOO ONKO NUMERO LIIAN ISO TAI LIIAN PIENI# 2
def testaus(arvattu):
    if arvattu in arvatutnumerot:
        print("[ EI SAMOJA! ]")
        return False
    if arvattu<maara and arvattu>-1:
        sopiva = True
        return True
    if arvattu-1 and sopiva==True:
        print("[ ONNISTUI! ]")
        return True

# KYSYY NUMEROITA NIIN KAUAN ETTÄ SAAT 7 VALID NUMEROA LISTAAN # 1
while len(arvatutnumerot)<arvattavamaara:
    try:
        print(f"ARVAA NUMERO [{laskin}]")
        arvattunumero = int(input(""))
        x = testaus(arvattunumero)
        if x==True:
            arvatutnumerot.append(arvattunumero)
            laskin+=1
    except:
        print("[ VIRHE! ]")
        continue

# GENEROI VALID NUMEROT RANDOMILLA JOTTA SAADAAN VOITTONUMEROT # 3
while len(oikeatnumerot)<arvattavamaara:
    generoidut = oikeatnumerot.append(random.randint(0,maara))
    if generoidut not in oikeatnumerot and generoidut!=None:
        oikeatnumerot.append(generoidut)
    else:
        continue

# LASKEE KUINKA MONTA ARVAUSTA MENI OIKEIN # 4
for i in range(len(arvatutnumerot)):
    if(arvatutnumerot[i] in oikeatnumerot):
        oikeinmenneet+=1


print("ARVOTAAN NUMEROT")
for i in range(len(oikeatnumerot)):
    print(f"[ SINUN LAPPU ]")
    print(arvatutnumerot)

    print(oikeatnumerot[i])
    time.sleep(0.25)
    os.system('cls')

print(f"[ SINUN LAPPU {arvatutnumerot}]")
print(f"[   JACKPOT   {oikeatnumerot}]")
print(f"[ SAIT {oikeinmenneet} NUMEROA OIKEIN! ]")
quit()

File: LottoSim.cs
// LottoSim in Csharp - Writen by Kaiku

List<int> Arvatut_Numerot = new List<int>();
List<int> Oikeat_Numerot = new List<int>();
int MaxMaara = 42;
int Arvattavat_Maara = 7;
int OikeinMenneet = 0;
int laskin = 1;

while(Arvatut_Numerot.Count != 7)
{
    Console.WriteLine("Numero " + laskin + "/" + Arvattavat_Maara);
   
    var x = int.Parse(Console.ReadLine());
    while(Arvatut_Numerot.Contains(x) || x > MaxMaara || x < 0) 
    { 
        Console.WriteLine("DUBLICATE OR TOO SMALL/LARGE"); 
        x = int.Parse(Console.ReadLine());
    }
    Arvatut_Numerot.Add(x);
    laskin++;
}
while(Oikeat_Numerot.Count != 7)
{
    var x = new Random().Next(0, MaxMaara);
    while (Oikeat_Numerot.Contains(x))
    {
        x = new Random().Next(0, MaxMaara);
    }
    Oikeat_Numerot.Add(x);
}
for(int i = 0; i < Oikeat_Numerot.Count; i++)
{
    if(Oikeat_Numerot.Contains(Arvatut_Numerot[i]))
    {
        OikeinMenneet++;
    }
}
Console.WriteLine("Sait " + OikeinMenneet + " numeroa oikein");